libraries/mbed/targets/hal/TARGET_Freescale/TARGET_K20XX/sleep.c

```diff
@@ -39,13 +39,27 @@ void deepsleep(void)
     SCB->SCR = 1<<SCB_SCR_SLEEPDEEP_Pos;
 
     __WFI();
-
     //Switch back to PLL as clock source if needed
     //The interrupt that woke up the device will run at reduced speed
     if (PLL_FLL_en) {
+        
+#if defined (TARGET_K20D50M)    
         if (MCG->C6 & (1<<MCG_C6_PLLS_SHIFT) != 0) /* If PLL */
             while((MCG->S & MCG_S_LOCK0_MASK) == 0x00U); /* Wait until locked */
         MCG->C1 &= ~MCG_C1_CLKS_MASK;
+#else    
+        MCG->C1 = (uint8_t)0x9Au;   
```

Comment:
That's no problem Martin I will do this, however I thought about this at the time and the original clock set up's are defined that way so I kept to the same convention, perhaps these should be updated as well, its a bit of a mixture there on all KLxx targets. Any thoughts why we need so much code to restart from deepsleep? or just go with what we have.